ui/src/main/js/global/channels/email/EmailGlobalConfiguration.js

```diff
@@ -0,0 +1,600 @@
+import React, { useState } from 'react';
+import * as PropTypes from 'prop-types';
+import CommonGlobalConfiguration from 'global/channels/CommonGlobalConfiguration';
+import { EMAIL_GLOBAL_FIELD_KEYS, EMAIL_INFO, EMAIL_TEST_FIELD } from 'global/channels/email/EmailModels';
+import CommonGlobalConfigurationForm from 'global/channels/CommonGlobalConfigurationForm';
+import TextInput from 'field/input/TextInput';
+import * as FieldModelUtilities from 'util/fieldModelUtilities';
+import CheckboxInput from 'field/input/CheckboxInput';
+import CollapsiblePane from 'component/common/CollapsiblePane';
+import NumberInput from 'field/input/NumberInput';
+import { CONTEXT_TYPE } from 'util/descriptorUtilities';
+
+const EmailGlobalConfiguration = ({ csrfToken, readonly }) => {
+    const [fieldModel, setFieldModel] = useState(FieldModelUtilities.createEmptyFieldModel([], CONTEXT_TYPE.GLOBAL, EMAIL_INFO.key));
+    const [fieldErrors, setFieldErrors] = useState({});
+    const [testFieldData, setTestFieldData] = useState({});
+
+    const handleChange = ({ target }) => {
+        const { type, name, value } = target;
+        const updatedValue = type === 'checkbox' ? target.checked.toString() : value;
+        const newState = Array.isArray(updatedValue) ? FieldModelUtilities.updateFieldModelValues(fieldModel, name, updatedValue) : FieldModelUtilities.updateFieldModelSingleValue(fieldModel, name, updatedValue);
+        setFieldModel(newState);
+    };
+
+    const handleTestChange = ({ target }) => {
+        const { type, name, value } = target;
+        const updatedValue = type === 'checkbox' ? target.checked.toString() : value;
+        const newState = { [name]: updatedValue };
+        setTestFieldData(newState);
+    };
+
+    const testField = (
+        <TextInput
+            key={EMAIL_TEST_FIELD.key}
+            name={EMAIL_TEST_FIELD.key}
+            label={EMAIL_TEST_FIELD.label}
+            description={EMAIL_TEST_FIELD.description}
+            onChange={handleTestChange}
+            value={testFieldData[EMAIL_TEST_FIELD.key]}
+        />
+    );
+
+    // FIXME collapsible pane needs to be open if data exists for the advanced section
+    return (
+        <CommonGlobalConfiguration
+            label={EMAIL_INFO.label}
+            description="Configure the email server that Alert will send emails to."
+            lastUpdated={fieldModel.lastUpdated}
+        >
+            <CommonGlobalConfigurationForm
+                csrfToken={csrfToken}
+                formData={fieldModel}
+                setErrors={(errors) => setFieldErrors(errors)}
+                setFormData={(content) => setFieldModel(content)}
+                testFields={testField}
+                testFormData={testFieldData}
+                setTestFormData={(values) => setTestFieldData(values)}
+            >
+                <TextInput
```

Comment:
Should we include ids in these fields?

Reply:
Upon further inspection, all the input fields look like they could use a shared key/ID value. Aparna has already opened a ticket for adding ID's to these items and I will make this change in the PR, adding a comment to that issue now.